mesonbuild/modules/rust.py

```diff
@@ -156,12 +156,16 @@ def test(self, state: ModuleState, args: T.Tuple[str, BuildTarget], kwargs: Func
         new_target_kwargs = base_target.original_kwargs.copy()
         # Don't mutate the shallow copied list, instead replace it with a new
         # one
-        new_target_kwargs['rust_args'] = \
-            new_target_kwargs.get('rust_args', []) + kwargs['rust_args'] + ['--test']
         new_target_kwargs['install'] = False
         new_target_kwargs['dependencies'] = new_target_kwargs.get('dependencies', []) + kwargs['dependencies']
         new_target_kwargs['link_with'] = new_target_kwargs.get('link_with', []) + kwargs['link_with']
 
+        lang_args = copy.copy(base_target.extra_args)
+        lang_args['rust'] = copy.copy(base_target.extra_args['rust'])
+        lang_args['rust'].extend(kwargs['rust_args'])
+        lang_args['rust'].append('--test')
```

Comment:
What's wrong with:
```suggestion
        lang_args['rust'] = base_target.extra_args['rust'] + kwargs['rust_args'] + ['--test']
```

It doesn't look meaningfully different from how we handle dependencies/link_with except in our case shorter since we don't use .get()